src/services/codefixes/importFixes.ts

```diff
@@ -432,22 +432,30 @@ namespace ts.codefix {
                     continue;
                 }
                 else if (indexOfStar !== -1) {
-                    const prefix = pattern.substr(0, indexOfStar);
-                    const suffix = pattern.substr(indexOfStar + 1);
+                    const prefix = removeLeadingDotSlash(pattern.substr(0, indexOfStar));
+                    const suffix = removeFileExtension(pattern.substr(indexOfStar + 1));
                     if (relativeName.length >= prefix.length + suffix.length &&
                         startsWith(relativeName, prefix) &&
                         endsWith(relativeName, suffix)) {
                         const matchedStar = relativeName.substr(prefix.length, relativeName.length - suffix.length);
-                        return key.replace("\*", matchedStar);
+                        return key.replace("*", matchedStar);
                     }
                 }
-                else if (pattern === relativeName || pattern === relativeNameWithIndex) {
-                    return key;
+                else {
+                    const cleanPattern = removeLeadingDotSlash(removeFileExtension(pattern));
+                    if (cleanPattern === relativeName || cleanPattern === relativeNameWithIndex) {
+                        return key;
+                    }
                 }
             }
         }
     }
 
+    /** Convert "./x" (and "././x") to "x". */
+    function removeLeadingDotSlash(path: string): string {
+        return startsWith(path, "./") || startsWith(path, ".\\") ? removeLeadingDotSlash(path.slice(2)) : path;

```

Comment:
the two relative paths are not relative to the same thing.. the input is relative to the current file, but the path is relative to the tsconfig.json. do not think removing the `./` is the right thing to do here. you probably want to expand both, or normalize one in terms of the other if it starts with a `./` or a `../`

Reply:
The two parameters to `tryGetModuleNameFromPaths` are relative to the baseUrl. I pushed a commit that updates their names.

Reply:
ahh.. good to know :)

what about  `..\`?

Reply:
Good catch, that needed a fix too.

Reply:
can u use getNormalizedPathComponents instead?

Reply:
Actually, this function is unused! #21092